components/src/core/DrawerLayout/DrawerLayout.tsx

```diff
@@ -25,11 +35,41 @@ const useStyles = makeStyles({
 });
 
 export const DrawerLayout: React.FC<DrawerLayoutProps> = (props) => {
-    const { children, drawer } = props;
+    const theme = useTheme();
+    const {
+        children,
+        drawer,
+        variantBreakpoints = {
+            [theme.breakpoints.down('xs')]: 'temporary',
+            [theme.breakpoints.up('sm')]: 'persistent',
+        },
+    } = props;
     const classes = useStyles(props);
+    const getDrawerContent = (): JSX.Element[] | JSX.Element => {
+        // if the drawer is a JSX.Element
+        const drawerElement = drawer as JSX.Element;
+        if (drawerElement.type && drawerElement.type.displayName === 'PXBlueDrawer') {
+            return Object.keys(variantBreakpoints).map(
+                (bp): JSX.Element =>
+                    useMediaQuery(bp) && <Drawer key={bp} variant={variantBreakpoints[bp]} {...drawerElement.props} />
+            );
+        }
+
+        // if the drawer is a React.FC
+        const drawerFC = drawer as React.FC<DrawerProps>;
+        return Object.keys(variantBreakpoints).map(
+            (bp): JSX.Element =>
+                useMediaQuery(bp) &&
+                drawerFC({
+                    key: bp,
+                    variant: variantBreakpoints,
+                    ...drawerElement.props,
+                })
+        );
```

Comment:
I think you need some more control logic here...if the user passes in a JSX.Element, but the displayName is not PXBlueDrawer, it falls back to this FC logic